services/venice-server/src/main/java/com/linkedin/venice/listener/grpc/handlers/GrpcOutboundStatsHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.linkedin.venice.listener.grpc.handlers;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.listener.ServerStatsContext;
+import com.linkedin.venice.listener.grpc.GrpcHandlerContext;
+import com.linkedin.venice.stats.ServerHttpRequestStats;
+import com.linkedin.venice.utils.LatencyUtils;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+
+public class GrpcOutboundStatsHandler extends VeniceServerGrpcHandler {
+  @Override
+  public void processNext(GrpcHandlerContext ctx) {
+    writeResponse(ctx);
+
+    // do not exit early if there is an error, we want to record stats for error cases as well
+    ServerStatsContext statsContext = ctx.getGrpcStatsContext();
+    HttpResponseStatus responseStatus = statsContext.getResponseStatus();
+    if (statsContext.getResponseStatus() == null) {
```

Comment:
IMO we should return error response instead of throwing an exception. We could additionally add a log statement